### PR TITLE
add xception ecndoer EncoderMixin

### PR DESCRIPTION
**Pull request by chlxydl**
Update Xception encoder with Mimin and depth support, local test OK.

### CODE REVIEW

File: segmentation_models_pytorch/encoders/xception.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+    def __init__(self, out_channels, depth=5, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        
+        self._out_channels = out_channels
+        self._depth = depth
+        self._in_channels = 3
+        #modify padding to maintain output shape
+        self.conv1 = nn.Conv2d(3, 32, 3, 2, 1, bias=False)
+        self.conv2 = nn.Conv2d(32, 64, 3, 1, 1, bias=False)
```
Comment (qubvel) [inline segmentation_models_pytorch/encoders/xception.py]: It is better to modify padding inplace, would be more clear imho
```python
self.conv1.padding = 1
self.conv2.padding = 1
```

File: segmentation_models_pytorch/encoders/xception.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        
+        self.pretrained = False
```
Comment (qubvel) [inline segmentation_models_pytorch/encoders/xception.py]: `pretrained` attribute is deprecated

File: segmentation_models_pytorch/encoders/xception.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
+import re
+import torch.nn as nn
+
+from pretrainedmodels.models.xception import pretrained_settings
+from pretrainedmodels.models.xception import Xception
+
+from ._base import EncoderMixin
+
+class XceptionEncoder(Xception,EncoderMixin):
```
[the remark this answers is not in the document]
Comment (chlxydl) [reply segmentation_models_pytorch/encoders/xception.py]: Fixed
